src/commands/diff.ts

```diff
@@ -125,13 +132,16 @@ export default class Diff extends Command {
         {
           override: overrides,
           outputType: outputFormat as diff.OutputType, // NOSONAR
+          markdownSubtype: markdownSubtype as diff.MarkdownSubtype
```

Comment:
I'm not 100% sure if this is correct. I can't figure what is recommended here https://clig.dev/

basically, if I do `asyncapi diff a.yml b.yml --type=json --markdownSubtype=yml` it will work and `markdownSubtype` is just ignored by diff library. 

now, is this right:
- throwing error that `markdownSubtype` is not supported in `json` type is kinda blocking for the user
- ignoring the type is good as diff still works, it's just that flag has no effect

maybe we should just display some kind of `info`?

we have similar case https://github.com/asyncapi/cli/blob/master/src/commands/generate/models.ts but not the same as there it makes sense to "error the user" because critical flag is missing

wdyt?

cc @Souvikns @magicmatatjahu

Reply:
I think displaying a"warning"(i.e. a text in yellow color) would be good. Like "Warning: Hey, this option won't work with the given type".

I think throwing an error is bad in this case. Like you said, it will be blocking for users.

Reply:
ok, can you make sure we display the warning then?

Reply:
@derberg Sorry for the long delay on this PR. Getting back to it now.

> ok, can you make sure we display the warning then?

How does this warning look?

![image](https://user-images.githubusercontent.com/54525741/213953069-61be2298-5e52-47cb-a5f0-2753b7fc0f1c.png)

Reply:
lgtm, maybe add also the value provided by user that will not work